### PR TITLE
Parse receiver files in C++ not in Fortran.

### PR DESCRIPTION
**Pull request by krenzland**
Add some tests. Delete unused code.

### CODE REVIEW

File: src/Reader/readpar.f90 [hunk 1]
```diff
@@ -2634,13 +2634,12 @@ SUBROUTINE readpar_output(EQN,DISC,IO,MPI,CalledFromStructCode)
       Refinement = 0
       pickdt = 0.1
       pickDtType = 1
-      nRecordPoints = 0
       energy_output_on = 0
       pickdt_energy = 1.0
       OutputRegionBounds(:) = 0.0
-!      RFileName = 'RecordPoints'
+      RFileName = ''
+      nRecordPoints = -1
```
Comment (git-taufiq) [inline src/Reader/readpar.f90:2641]: What will happen when we set nRecordPoints = -1?

**Review by git-taufiq — COMMENTED** [review]
src/Reader/readpar.f90:2641 What will happen when we set nRecordPoints = -1?

**Review by Thomas-Ulrich — APPROVED** [review]
looks good.

File: src/ResultWriter/ReceiverWriter.cpp [hunk 2]
```diff
@@ -47,6 +47,44 @@
...
+  for (; iter != end; ++iter, ++numberOfCoordinates) {
+    if (numberOfCoordinates > coordinates.size()) {
+      break;
+    }
+    coordinates[numberOfCoordinates] = std::stod(*iter);
+  }
+  if (numberOfCoordinates != coordinates.size()) {
+    throw std::runtime_error("Incorrect number of coordinates in line " + line + ".");
+  }
+  return {coordinates[0], coordinates[1], coordinates[2]};
```
Comment (sebwolf-de) [inline src/ResultWriter/ReceiverWriter.cpp:73]: Why do you have the `std::array` as intermediate representation and not an `Eigen::Vector3d` from the beginning?

File: src/ResultWriter/ReceiverWriter.cpp [hunk 3]
```diff
@@ -47,6 +47,44 @@
...
+}
+
+std::vector<Eigen::Vector3d> seissol::writer::parseReceiverFile(const std::string& receiverFileName) {
+  std::vector<Eigen::Vector3d> points{};
+
+  std::ifstream file{receiverFileName};
+  std::string line{};
+  while (std::getline(file, line)) {
+    bool onlyWhiteSpace = std::all_of(line.begin(), line.end(), isspace);
```
Comment (sebwolf-de) [inline src/ResultWriter/ReceiverWriter.cpp:82]: Can you make it more verbose: `std::isspace`?

File: src/tests/ResultWriter/ReceiverWriter.t.h [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#include "ResultWriter/ReceiverWriter.h"
+namespace seissol::unit_test {
+
+TEST_CASE("Parses line correctly") {
+  const auto lines = std::vector<std::string>{
+  "1\t0.1  10",
+  "10\t2\t0.2",
+  "1e4   1e3  1e-10"
```
Comment (sebwolf-de) [inline src/tests/ResultWriter/ReceiverWriter.t.h:8]: Maybe add some negative values, just to make the tests more complete.

**Review by sebwolf-de — COMMENTED** [review]
In general great changes, just some minor improvements...

File: src/Reader/readpar.f90 (hunk 1 above)
Comment (krenzland) [reply src/Reader/readpar.f90:2641]: Nothing. It's unused - it's just an easy hack to see whether it was used in the parameters.par - if yes, it issues a warning

File: src/ResultWriter/ReceiverWriter.cpp (hunk 3 above)
Comment (krenzland) [reply src/ResultWriter/ReceiverWriter.cpp:82]: Unfortunately, this is just more ugly:
https://stackoverflow.com/a/46208892
std::isspace is overloaded, isspace not. Not sure why...

File: src/ResultWriter/ReceiverWriter.cpp (hunk 2 above)
Comment (krenzland) [reply src/ResultWriter/ReceiverWriter.cpp:73]: A very good question indeed...

**Review by sebwolf-de — APPROVED** [review]

**Pull request merged**